typescript/mainLogic/services/service_context.ts

```diff
@@ -0,0 +1,157 @@
+/*package services*/ //TODO Get rid of this, not needed in typescript
+
+//import (
+	//"context" //TODO Don't need context for typescript, golang specific
+import * as core_api_bindings_js_grpc from '../../core_api_bindings/api_container_service_grpc_pb.js';
+import core_api_bindings_js from '../../core_api_bindings/api_container_service_pb.js'; //TODO - extra line, I might be able to reduce this with line above
+//	"github.com/palantir/stacktrace" //import stacktrace = require('github.com/palantir/stacktrace'); // TODO - Don't need stackTrace for golang
+	var path = require("path"); //"path" - TODO importing from node.js potentially
+//)
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+interface GeneratedFileFilepaths { //TODO Chose to use interface to represent struct, will need to test to see if this hypothesis works
+	AbsoluteFilepathOnTestsuiteContainer: string;
+	AbsoluteFilepathOnServiceContainer:   string;
+}
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+interface ServiceContext {
+	client:                                   core_api_bindings_js_grpc.ApiContainerServiceClient;
+	serviceId:                                ServiceID;
+	ipAddress:                                string;
+	testVolumeMountpointOnTestsuiteContainer: string;
+	testVolumeMountpointOnServiceContainer:   string;
+}
+
+function NewServiceContext (
```

Comment:
Some comment about capitalization - in Typescript, I think functions are never uppercased (Go is just weird in this regard)

Reply:
Also, this is another language-specific thing - Go doesn't have constructors so the way to do it is have a random `NewXXXXXX` function. Typescript does have constructors, so we can move this into a constructor inside the `ServiceContext` class